xdsl/ir/affine/affine_map.py

```diff
@@ -42,6 +55,43 @@ def transpose_map() -> AffineMap:
     def empty() -> AffineMap:
         return AffineMap(0, 0, ())
 
+    @staticmethod
+    def from_callable(
+        func: AffineMapBuilderT, *, dim_symbol_split: tuple[int, int] | None = None
+    ) -> AffineMap:
+        """
+        Creates an `AffineMap` by calling the function provided. If `dim_symbol_split` is
+        not provided or `None`, then all parameters are treated as dimension expressions.
+        If `dim_symbol_split` is provided, `func` is expected to have the same number of
```

Comment:
Does this condition/expectation need to be tested within this or do we just let all error handling/checking trickle down to `AffineMap`?

Reply:
I was thinking about that. It would be really easy to test in the code, and I didn't do it because it feels like it'll be slow but, thinking about it some more, if that's a problem then the user can use a different initializer, so it shouldn't really be a concern. I'll add the check.